scripts/fuse/README.MD

```diff
@@ -0,0 +1,166 @@
+
+These first two layers, (e.g `mountpoint/12/system:`) are read only. Deeper layers support writing: any change is directly updated in the backing Elektra-Key-Database.
+
+### Process Context Mocking
+
+All Elektra operations are performed "from the perspective" of the respective process.
+This is done by only calling libelektra/kdb from special "mock"-processes that mimic all of the relevant attributes Elektra uses to resolve its keys.
+
+The mocked attributes are:
+- the user and group (for the `user:` namespace)
+- working directory (for the `dir:` namespace)
+- environment variables (for the `proc:` namespace)
+- in the future: the process arguments
+
+
+### Example file system structure
+
+The filesystem structure mounted below the mountpoint could present as follows: (restructed to only one process with pid 41, with a maximum listing depth of 4)
+
+```
+|-- 41
+|   |-- cascading:
+|   |   |-- dir_and_file_at_once
+|   |   |   `-- leaf
+|   |   |-- dirkey
+|   |   |-- elektra
+|   |   |   |-- modules
+|   |   |   `-- version
+|   |   |-- info
+|   |   `-- person
+|   |       `-- name
+|   `-- user:
+|       |-- dir_and_file_at_once
+|       |   |-- @elektra.value
+|       |   `-- leaf
+|       |-- info
+|       `-- person
+|           `-- name
+
+```
+as obtained with
+```sh
+tree -L 4 mountpoint
+```
+
+
+
+## Assumptions and special behaviour
+
+Elektra does not map perfectly to the structure of a classical filesystem.
+Whenever deviations are necessary, the goal is to maximize usability and adhere to the expectations of common unix tools.
+
+### Intermediate directories
+
+In Elektra, there is no need to have a `path` of continous keys between ancestors.
+E.g. the existence of `user:/a` and `user:/a/b/c` does not imply the existence of `user:/a/b`.
+Such intermediate keys are mapped to "virtual" directories, s.t. navigation in an interactive manner is possible and the directory tree is continous.
+
+A side effect is that creating/deleting a single key can create/delete multiple "virtual" directories at once.
+Another is that copies created with `cp -r` will not have these vacuous intermediaries but instead empty keys.
+
+### Keys that act both as file and directory
+
+In Elektra, there are only keys and no distinction between directories and files are made. To enable a bidirectional mapping into a classical file system, the following mechanism is employed:
+
+If a key (`A`) exists, and has some child key (`B`), `A` is treated as a directory containing `B` and a special file `@elektra.value`. This file is used to access the value of `A`.
+
+Some side effects arise:
+E.g. recursively deleting directories with `rm -r` containing `@elektra.value` files raises (task uncritical) errors when attempting to delete a parent directory which was already deleted with the removal of an `@elektra.value` file directly below.
```

Comment:
```suggestion
E.g. recursively deleting directories with `rm -r` containing `®elektra.value` files raises (task uncritical) errors when attempting to delete a parent directory which was already deleted with the removal of an `®elektra.value` file directly below.
```

see #3599

Reply:
>Some (older) file systems may have trouble with the ® character (or any other non-ASCII characters).

As we implement a file system here, this should not be a problem. Terminals or language settings not supporting the ® character, however, can be a problem. In specific, the fuse Docker image we provide doesn't have locales installed, so ® does not work (at least when I tried the last time).

In any case, we should follow our own standard, so either, we

1. use ®elektra
2. also reserve ®elektra.value (or even ®elektra.*)

Reply:
I would suggest we use the `®elektra` folder if possible. From a general point of view it makes more sense for Elektra to only reserve `®elektra` as a key name part. Since the KDB is hierarchical anyway there is no need to reserve all key name parts that start with X.

Reply:
>    reduces usability (one file versus one directory with always one file in it)

Yes, it is pressing 2x tab instead of one. The question is if there is a possibility that more files get added in the future, e.g. for exporting functionality?

> Furthermore, again because of usability considerations, I'm not happy with the use of ® instead of @ due to the mentioned encoding problems. Imho, any such naming clash is extremely unlikely in practical use cases, whereas the pain of obfuscated encodings is real.

I see your point. Is there also a possibility of obfuscated encodings except if the locales package is not installed at all? We chose ® exactly because it also has somewhat reasonable encoding if the wrong locale is chosen (even in ASCII).

@kodebach there is still a "TODO: link to detailed ®elektra document" in doc/KEYNAMES.md. Would be better to have this document then long discussions here in this sub-thread never to be found again. We can continue the discussion in the PR.

I am open to `@elektra` but then `@elektra` and not `®elektra` should be the reserved key name.

Btw. I didn't see the initial response https://github.com/ElektraInitiative/libelektra/pull/3657#discussion_r604835066 of @dev2718 when I wrote https://github.com/ElektraInitiative/libelektra/pull/3657#issuecomment-814239192

Reply:
As of f13604c3f71c210cbc9dbc115069d96ec50ba9ad `®elektra` is standard, but configurable via command line option.